Reachable unwrap panic in `FdDirOps::new_inode` when accessing an exiting process's file descriptors

### Describe the bug
There is a reachable unwrap panic in `IdleClassRq::enqueue()` at kernel/src/fs/procfs/pid/fd.rs:33 due to a race condition when accessing a process's file descriptors in procfs simultaneously as the process is exiting.

https://github.com/asterinas/asterinas/blob/4a9977d9a7d071dbe2ad4cea8ae69d4db81f566e/kernel/src/fs/procfs/pid/fd.rs#L30-L33

### To Reproduce
1. Compile the program and run:
```C
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>
#include <sys/xattr.h>
#include <string.h>
#include <pthread.h>
#include <sched.h>
#include <stdatomic.h>
#include <signal.h>

atomic_int start_hammering = 0;
atomic_int stop_hammering = 0;

void *access_proc_fd(void *arg) {
    char *path = (char *)arg;
    char list[1024];

    while (atomic_load(&start_hammering) == 0) {
        // spin
    }

    while (atomic_load(&stop_hammering) == 0) {
        llistxattr(path, list, sizeof(list));
        sched_yield();
    }
    return NULL;
}

int main() {
    pid_t pid = fork();

    if (pid < 0) {
        perror("fork");
        return 1;
    }

    if (pid == 0) {
        while(1) {
            sleep(1);
        }
        exit(0);
    } else {
        char path[256];
        snprintf(path, sizeof(path), "/proc/%d/fd", pid);

        usleep(100000);

        const int NUM_THREADS = 4;
        pthread_t threads[NUM_THREADS];

        for (int i = 0; i < NUM_THREADS; i++) {
            if (pthread_create(&threads[i], NULL, access_proc_fd, path) != 0) {
                perror("pthread_create");
                kill(pid, SIGKILL);
                waitpid(pid, NULL, 0);
                return 1;
            }
        }

        atomic_store(&start_hammering, 1);
        usleep(50000);
        kill(pid, SIGKILL);
        waitpid(pid, NULL, 0);
        atomic_store(&stop_hammering, 1);

        for (int i = 0; i < NUM_THREADS; i++) {
            pthread_join(threads[i], NULL);
        }
    }

    return 0;
}
```

### Expected behavior
Asterinas shall not panic when run the program.

### Environment
- Official docker asterinas/asterinas:0.15.0-20250516
- Intel(R) Xeon(R) Gold 6230R CPU @ 2.10GHz
- Asterinas version: main 4a9977d

### Log
```
~ # /root/llistxattr.c 
[    10.054] ERROR: Uncaught panic:
        called `Option::unwrap()` on a `None` value
        at /root/asterinas/kernel/src/fs/procfs/pid/fd.rs:33
        on CPU 0 by thread Some(Thread { task: (Weak), data: Any { .. }, status: AtomicThreadStatus(1), cpu_affinity: AtomicCpuSet { bits: [1] }, sched_attr: SchedAttr { policy: SchedPolicyState { kind: AtomicSchedPolicyKind(2), policy: UnsafeCell { .. } }, last_cpu: AtomicCpuId(0), real_time: RealTimeAttr { prio: 99, time_slice: 31490040 }, fair: FairAttr { weight: 1024, vruntime: 196198243 } } })
Printing stack trace:
   1: fn 0xffffffff88829910 - pc 0xffffffff8882995a / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffec9ede0;

   2: fn 0xffffffff8813db50 - pc 0xffffffff8813e2f9 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffec9ee30;

   3: fn 0xffffffff8813db40 - pc 0xffffffff8813db4a / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffec9f240;

   4: fn 0xffffffff8804b000 - pc 0xffffffff8804b00a / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffec9f250;

   5: fn 0xffffffff88943400 - pc 0xffffffff88943444 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffec9f260;

   6: fn 0xffffffff889434b0 - pc 0xffffffff889434f0 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffec9f2a0;

   7: fn 0xffffffff8894df20 - pc 0xffffffff8894df35 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffec9f310;

   8: fn 0xffffffff8849b940 - pc 0xffffffff8849bae2 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffec9f320;

   9: fn 0xffffffff88438280 - pc 0xffffffff884383da / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffec9f470;

  10: fn 0xffffffff88317f00 - pc 0xffffffff8831808b / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffec9f530;

  11: fn 0xffffffff880c8e90 - pc 0xffffffff880c8f22 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffec9f680;

  12: fn 0xffffffff880cb6a0 - pc 0xffffffff880cb9cf / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffec9f800;

  13: fn 0xffffffff880ab420 - pc 0xffffffff880ab86f / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffec9f9d0;

  14: fn 0xffffffff880aa720 - pc 0xffffffff880aa7b7 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffec9fd50;

  15: fn 0xffffffff880aa6b0 - pc 0xffffffff880aa705 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffeca02c0;

  16: fn 0xffffffff8810ff50 - pc 0xffffffff8811017a / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffeca0340;

  17: fn 0xffffffff880a0f80 - pc 0xffffffff880a1067 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffeca0560;

  18: fn 0xffffffff8810a980 - pc 0xffffffff8810aab5 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffeca0680;

  19: fn 0xffffffff8810a270 - pc 0xffffffff8810a373 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffeca0810;

  20: fn 0xffffffff88388030 - pc 0xffffffff883a9736 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffeca0980;

  21: fn 0xffffffff8813d890 - pc 0xffffffff8813d921 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffecb2510;

  22: fn 0xffffffff88110c30 - pc 0xffffffff8811140a / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffecb2690;

  23: fn 0xffffffff8852bc60 - pc 0xffffffff8852bc66 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffecb2d00;

  24: fn 0xffffffff8856d770 - pc 0xffffffff8856d787 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffecb2d10;

  25: fn 0xffffffff8856d6d0 - pc 0xffffffff8856d6f1 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffecb2d30;

  26: fn 0xffffffff88305eb0 - pc 0xffffffff88305ec2 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffecb2d80;

  27: fn 0xffffffff8804bce0 - pc 0xffffffff8804bcf6 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffecb2db0;

  28: fn 0xffffffff880c83c0 - pc 0xffffffff880c83cc / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffecb2f50;

  29: fn 0xffffffff8852ace0 - pc 0xffffffff8852acee / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffecb2f80;

  30: fn 0xffffffff887ebf50 - pc 0xffffffff887ebf75 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffecb2fa0;

  31: fn 0xffffffff887e75b0 - pc 0xffffffff887e7649 / registers:

     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffecb2fd0;


     rax 0xffffffff88c7ed18; rdx  0x100000000000008; rcx 0xffffffff88c7ed18; rbx                0x0;
     rsi 0xffffdffffec9ed60; rdi 0xffffffff887bd281; rbp                0x0; rsp 0xffffdffffecb3000;

OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panic.rs:107
(  2) /root/asterinas/kernel/src/thread/oops.rs:129
(  3) ??:?
(  4) bb0e31ooocbssppv6q3sbsuao:?
(  5) ??:?
(  6) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:217
(  7) ??:?
(  8) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:977
(  9) /root/asterinas/kernel/src/fs/procfs/pid/mod.rs:64
( 10) /root/asterinas/kernel/src/fs/procfs/template/dir.rs:170
( 11) /root/asterinas/kernel/src/fs/path/dentry.rs:182
( 12) /root/asterinas/kernel/src/fs/path/dentry.rs:564
( 13) /root/asterinas/kernel/src/fs/fs_resolver.rs:238
( 14) /root/asterinas/kernel/src/fs/fs_resolver.rs:164
( 15) /root/asterinas/kernel/src/fs/fs_resolver.rs:158
( 16) /root/asterinas/kernel/src/syscall/setxattr.rs:137
( 17) /root/asterinas/kernel/src/syscall/setxattr.rs:146
( 18) /root/asterinas/kernel/src/syscall/listxattr.rs:92
( 19) /root/asterinas/kernel/src/syscall/listxattr.rs:46
( 20) /root/asterinas/kernel/src/syscall/mod.rs:194
( 21) /root/asterinas/kernel/src/syscall/mod.rs:351
( 22) /root/asterinas/kernel/src/thread/task.rs:83
( 23) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:250
( 24) /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.5/src/panicking.rs:56
( 25) /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.5/src/panicking.rs:42
( 26) /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.5/src/panic.rs:87
( 27) /root/asterinas/kernel/src/thread/oops.rs:54
( 28) /root/asterinas/kernel/src/thread/task.rs:109
( 29) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:250
( 30) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2016
( 31) /root/asterinas/ostd/src/task/mod.rs:234
make: *** [Makefile:230: run] Error 1
```

--- a/kernel/src/fs/procfs/pid/fd.rs
+++ b/kernel/src/fs/procfs/pid/fd.rs
@@ -27,12 +27,13 @@ impl FdDirOps {
             .parent(parent)
             .build()
             .unwrap();
-        file_table
-            .lock()
-            .as_ref()
-            .unwrap()
-            .read()
-            .register_observer(Arc::downgrade(&fd_inode) as _);
+        // Guard against the race condition when procfs is being accessed for an exiting process,
+        // whose file table may have already been released.
+        if let Some(file_table_ref) = file_table.lock().as_ref() {
+            file_table_ref
+                .read()
+                .register_observer(Arc::downgrade(&fd_inode) as _);
+        }
 
         fd_inode
     }

